server/api/crud/artifacts.py

```diff
@@ -301,15 +305,24 @@ def _delete_artifact_data(
                 object_uid=object_uid,
             )
 
-            # Data artifacts that are ModelArtifact, DirArtifact, or DatasetArtifact
-            # must not be removed because we do not yet support the deletion of artifacts that contain multiple files
+            path = artifact["spec"]["target_path"]
+
+            # Data artifacts that are ModelArtifact, DirArtifact must not be removed because we do not yet
+            # support the deletion of artifacts that contain multiple files
+            # We support deleting DatasetArtifact data that contains one file
             # TODO: must be removed once it is supported
             artifact_kind = artifact["kind"]
-            if artifact_kind in ["model", "dataset", "dir"]:
+            if artifact_kind in ["model", "dir"]:
                 raise mlrun.errors.MLRunNotImplementedServerError(
                     f"Deleting artifact data kind: {artifact_kind} is currently not supported"
                 )
-            path = artifact["spec"]["target_path"]
+            if artifact_kind == "dataset" and not self._is_artifact_data_single_file(
+                path
+            ):
+                raise mlrun.errors.MLRunNotImplementedServerError(
+                    "Deleting artifact data kind dataset is currently available for a single file"
+                )
+
```

Comment:
```suggestion
            ):
                raise mlrun.errors.MLRunNotImplementedServerError(
                    "Deleting artifact data of kind 'dataset' is currently supported for a single file only"
                )

```

Reply:
The error message will be defined by @jillnogold  - https://iguazio.atlassian.net/browse/ML-7070

Reply:
@jillnogold could you please take a look here?

Reply:
@rokatyy She knows that, the ticket is on here.